src/pybind/mgr/cephadm/module.py

```diff
@@ -1975,6 +1975,9 @@ def remove_service(self, service_name: str, force: bool = False) -> str:
             if self.spec_store[service_name].spec.service_type in ('mon', 'mgr'):
                 return f'Unable to remove {service_name} service.\n' \
                        f'Note, you might want to mark the {service_name} service as "unmanaged"'
+        else:
+            raise OrchestratorError(f"Invalid service '{service_name}'. Use 'ceph orch ls' "
```

Comment:
http://pulpito.front.sepia.ceph.com/adking-2022-03-21_05:58:33-orch:cephadm-wip-adk-testing-2022-03-20-2156-distro-basic-smithi/6749700/

```
2022-03-22T00:06:50.128 DEBUG:teuthology.orchestra.run.smithi110:> sudo adjust-ulimits ceph-coverage /home/ubuntu/cephtest/archive/coverage timeout 120 ceph --cluster ceph nfs cluster rm test
2022-03-22T00:06:50.470 INFO:teuthology.orchestra.run.smithi110.stderr:Error EINVAL: Invalid service 'ingress.nfs.test'. Use 'ceph orch ls' to list available services.
2022-03-22T00:06:50.470 INFO:teuthology.orchestra.run.smithi110.stderr:
2022-03-22T00:06:50.474 DEBUG:teuthology.orchestra.run:got remote process result: 22
2022-03-22T00:06:50.475 DEBUG:teuthology.orchestra.run.smithi110:> sudo adjust-ulimits ceph-coverage /home/ubuntu/cephtest/archive/coverage timeout 120 ceph --cluster ceph log 'Ended test tasks.cephfs.test_nfs.TestNFS.test_cluster_info'
2022-03-22T00:06:50.892 INFO:journalctl@ceph.mgr.a.smithi110.stdout:Mar 22 00:06:50 smithi110 bash[34814]: debug 2022-03-22T00:06:50.460+0000 7f6eb9f02700 -1 log_channel(cephadm) log [ERR] : Invalid service 'ingress.nfs.test'. Use 'ceph orch ls' to list available services.
2022-03-22T00:06:50.893 INFO:journalctl@ceph.mgr.a.smithi110.stdout:Mar 22 00:06:50 smithi110 bash[34814]: Traceback (most recent call last):
2022-03-22T00:06:50.893 INFO:journalctl@ceph.mgr.a.smithi110.stdout:Mar 22 00:06:50 smithi110 bash[34814]:   File "/usr/share/ceph/mgr/orchestrator/_interface.py", line 125, in wrapper
2022-03-22T00:06:50.893 INFO:journalctl@ceph.mgr.a.smithi110.stdout:Mar 22 00:06:50 smithi110 bash[34814]:     return OrchResult(f(*args, **kwargs))
2022-03-22T00:06:50.893 INFO:journalctl@ceph.mgr.a.smithi110.stdout:Mar 22 00:06:50 smithi110 bash[34814]:   File "/usr/share/ceph/mgr/cephadm/module.py", line 1997, in remove_service
2022-03-22T00:06:50.893 INFO:journalctl@ceph.mgr.a.smithi110.stdout:Mar 22 00:06:50 smithi110 bash[34814]:     raise OrchestratorError(f"Invalid service '{service_name}'. Use 'ceph orch ls' "
2022-03-22T00:06:50.894 INFO:journalctl@ceph.mgr.a.smithi110.stdout:Mar 22 00:06:50 smithi110 bash[34814]: orchestrator._interface.OrchestratorError: Invalid service 'ingress.nfs.test'. Use 'ceph orch ls' to list available services.
2022-03-22T00:06:50.894 INFO:journalctl@ceph.mgr.a.smithi110.stdout:Mar 22 00:06:50 smithi110 bash[34814]: debug 2022-03-22T00:06:50.464+0000 7f6eb9f02700 -1 mgr.server reply reply (22) Invalid argument Invalid service 'ingress.nfs.test'. Use 'ceph orch ls' to list available services.
```
I think I'm remembering the reason this wasn't implemented previously. I think we wanted this command to be idempotent so situations like this didn't cause errors or break tests. Can you have this still return code 0 while still giving the same message that the service wasn't found (probably by just returning here instead of throwing an exception)